src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -177,15 +110,84 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 
 	activate(xterm: Terminal): void {
 		this._terminal = xterm;
-		this._register(xterm.parser.registerOscHandler(ShellIntegrationOscPs.VSCode, data => {
-			return this._handleVSCodeSequence(data);
-		}));
-		this._register(xterm.onData(e => {
+		this._register(xterm.onData(async e => {
 			this._lastUserData = e;
-			this._lastUserDataTimestamp = Date.now();
 		}));
 	}
 
+	private async _handleExtensionProviders(terminal?: Terminal): Promise<void> {
```

Comment:
We should name this `_handleCompletionProviders` as it includes more than the soon to be extension providers